src/pages/Post.js

```diff
@@ -9,37 +10,56 @@ import Modal from "../components/Modal";
 import Header from "../components/Header";
 
 export default function PostPage() {
-  const [userData, setUserData] = useState({ data: null });
-  const [questionList, setQuestionList] = useState({ data: null });
-  const [isModalOpen, setIsModalOpen] = useState(false);
-  const modalBackgroundRef = useRef();
+    const [userData, setUserData] = useState({ data: null });
+    const [questionList, setQuestionList] = useState({ data: null });
+    const [isModalOpen, setIsModalOpen] = useState(false);
+    const modalBackgroundRef = useRef();
+    const params = useParams();
 
-  useEffect(() => {
     async function fetchData() {
-      const userData = await getUserData(5739);
-      const questionList = await getQuestionList(5739);
-      setUserData(userData);
-      setQuestionList(questionList);
+        try {
+            const userData = await getUserData(params.id);
+            const questionList = await getQuestionList(params.id);
```

Comment:
이왕에 params 쓰는김에 이렇게 쓰는게 더 괜찮을것 같네요

```suggestion
    const { id } = useParams();

    async function fetchData() {
        try {
            const userData = await getUserData(id);
            const questionList = await getQuestionList(id);
```